kube-client/src/client/auth/oauth.rs

```diff
@@ -111,14 +111,25 @@ impl Gcp {
             Ok(TokenOrRequest::Request {
                 request, scope_hash, ..
             }) => {
-                #[cfg(not(any(feature = "rustls-tls", feature = "openssl-tls")))]
+                #[cfg(not(any(
+                    all(
+                        feature = "rustls-tls",
+                        any(feature = "ring", feature = "aws-lc-rs", feature = "webpki-roots")
```

Comment:
i think we need;
- one of `aws-lc-rs` and `ring`
- `webpki-roots` optional, but orthogonal to this choice (so probably does not need to be checked here)

Reply:
see my previous comment.
let's try adding cfg-if to make it a bit clearer

Reply:
i think we can avoid this extra change and dependency, and keep all logic as we had it before by introducing the new compile error up to top level (257 in oidc.rs).

```rust
#[cfg(all(feature = "rustls-tls", not(any(feature = "ring", feature = "aws-lc-rs"))))]
compile_error!("At least one of ring or aws-lc-rs feature must be enabled to use rustls-tls feature")
```

Reply:
mostly because the way it's indented now is also quite big and hard to extend, and we want to just bail out as soon as possible anyway.

besides, oidc.rs isn't even the client that is used for most things (that one comes from config_ext.rs and does the selection in [tls.rs](https://github.com/kube-rs/kube/blob/main/kube-client/src/client/tls.rs#L50-L70) (which does not do this same if-else).

Reply:
done.

btw, are you fine with removing the [auto-installation](https://github.com/kube-rs/kube/blob/main/kube-client/src/client/builder.rs#L99) of the aws-lc-rs provider if aws-lc-rs feature was set?
theoretically both ring and aws-lc-rs might be enabled, and we should leave it to the user (or let [rustls](https://github.com/rustls/rustls/blob/a209a0e805ed2391eb4e0f3b3ed00b50ed3f4829/rustls/src/crypto/mod.rs#L261) install if only one of them was selected)

Reply:
> removing auto-installation

not super comfortable touching this yet. removing it outright will feel breaking, and it's a very low level api people will have to familiarise themselves with to do basic things when using aws-lc-rs. if aws-lc-rs becomes the default, i would want it to be frictionless, and this feels ok to me as a 'helping hand'; install if it hasn't been installed already. nothing stops people from installing another default provider else earlier on.

this might turn into a bigger discussion though. maybe we need to do the same thing we did for openssl vs rustls; always speculatively install one provider, tilting towards our default. that way we can change our default later on (like we did for openssl -> rustls).